packages/checkbox/tests/checkbox.test.tsx

```diff
@@ -517,3 +517,47 @@ test("Uncontrolled FormControl - calls all onBlur EventHandler", () => {
   expect(formControlOnBlurMock).toHaveBeenCalled()
   expect(checkboxOnBlurMock).toHaveBeenCalled()
 })
+
+test("On reseting form, checkbox should reset to its default state i.e., checked", () => {
+  const Component = () => (
+    <form
+      onSubmit={(e) => {
+        e.preventDefault()
+        e.currentTarget.reset()
+      }}
+    >
+      <label htmlFor="myCheckbox">My Checkbox</label>
+      <Checkbox id="myCheckbox" defaultChecked />
+      <button type="submit">Reset</button>
```

Comment:
I think we can replace the button here to use `type=reset`. That way you can remove the `onSubmit` handler completely.

Reply:
done